### ISSUE TITLE
Support for `Option<&T>`

### ISSUE THREAD

**Issue by oguzkocer** [mozilla/uniffi-rs#2149]
According to the [documentation](https://mozilla.github.io/uniffi-rs/0.28/proc_macro/index.html#the-uniffiexport-attribute)([source](https://github.com/mozilla/uniffi-rs/blob/v0.28.0/docs/manual/src/proc_macro/index.md?plain=1#L109-L112)), `Option<&T>` is supported by proc macro:

```
#[uniffi::export]
fn process_data(a: &MyRecord, b: &MyEnum, c: Option<&MyRecord>) {}
```

However, this seems to result in the following error: 

```
| fn process_data(_a: &MyRecord, _b: &MyEnum, _c: Option<&MyRecord>) {}
|                                                 ^^^^^^^^^^^^^^^^^ the trait `uniffi::TypeId<UniFfiTag>` is not implemented for `&MyRecord`, which is required by `Option<&MyRecord>: uniffi::TypeId<UniFfiTag>`
```

[Here is a demo project](https://github.com/oguzkocer/uniffi_option_ref_t) that demonstrates this. Hopefully I am missing something, but if not, would it be possible to add support for this?

The alternative at the moment is to use `&Option<T>`, but that has completely different semantics to `Option<&T>`.

**Comment by bendk** [comment]
Yeah, that docstring has been incorrect since it was added.  I'm not sure why I wrote that, I think I meant to say `&Option<T>`.  Supporting `Option<&T>` would be quite tricky.  I think it would mean generalizing the lifting code so that references weren't special cased.  I think this would mean some sort of 2-phase system that would be a significant increase in complexity.

For now, I'm going to fix the docstring to describe the current truth.  Maybe we could eventually handle this though.
